task-driver/src/settle_match_internal.rs

```diff
@@ -244,14 +258,46 @@ impl SettleMatchInternalTask {
             global_state,
             proof_manager_work_queue,
             task_state: SettleMatchInternalTaskState::Pending,
+        };
+
+        // Try to lock both wallets, if they cannot be locked then the task cannot be run
+        // and the internal matching engine will re-run next time the proofs are updated
+        let wallet1 = self_.find_wallet_for_order(&order1).await?;
+        let wallet2 = self_.find_wallet_for_order(&order2).await?;
+
+        if !wallet1.try_lock_wallet() {
```

Comment:
not sure how to think about this, but is this the only place where we'll have to lock a wallet? i'm sure there's many other places in the code where we find a wallet

Reply:
well the only places you'd do so are where you update a wallet, which is in the tasks.